### PR TITLE
docs(react-integration): revamp into complete guide

### PR DESCRIPTION
**Pull request by tanner-reits**
Updates the docs for the React framework wrapper to follow the same structure as the Angular docs

### CODE REVIEW

File: src/docs/framework-integration/react.md
Comment (rwaskiewicz) [inline src/docs/framework-integration/react.md:231]: Optional word smithing to use some of the same nomenclature as the rest of the doc:
```suggestion
The name of the Stencil package where components are available for consumers (i.e. the value of the `name` property in your Stencil component library's `package.json`).
```

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]

File: src/docs/framework-integration/react.md
Comment (tanner-reits) [reply src/docs/framework-integration/react.md:303]: I think this is just an issue with the `dist-custom-elements` and `dist-custom-elements-bundle` target terminologies being used too loosely in the past. I'll update this to not say "Bundle" but keep the link the same. However, there may be some additional cleanup we need to address for V3 with the `loaderDir` config option since that is expecting a `defineCustomElements` function that would only exist if `customElementsExportBehavior: 'bundle'` is set.

File: src/docs/framework-integration/react.md
Comment (tanner-reits) [reply src/docs/framework-integration/react.md:231]: @rwaskiewicz I like this change, but this same thing probably needs to happen in the Angular and Vue docs as well. So, I'll hold off on making this swap until we merge all these back into one PR so then I can just change them all at once

File: src/docs/framework-integration/react.md
Comment (rwaskiewicz) [reply src/docs/framework-integration/react.md]: > I'll update this to not say "Bundle" but keep the link the same. 

👍 

> However, there may be some additional cleanup we need to address for V3 with the loaderDir config option since that is expecting a defineCustomElements function that would only exist if customElementsExportBehavior: 'bundle' is set.

Can you capture those TODOs somewhere so they don't fall through the cracks?

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]

File: versioned_docs/version-v2/framework-integration/react.md
Comment (tanner-reits) [reply versioned_docs/version-v2/framework-integration/react.md:298]: First off, I decided to swap this off of `create-react-app` to a Vite React app instead in [4d88bc3](https://github.com/ionic-team/stencil-site/pull/964/commits/4d88bc3a9b31d628edce650bdb62cbb66793c77d). Did this for two reasons:
1. The command for generating w/ `create-react-app` has some issues in Yarn environments
2. `create-react-app` is kinda dead, so it's easy enough for us to move away from it

Even so, we can't use `npm2yarn` here because that plugin can't correctly convert certain syntaxes (like `npm create`/`npm init`) to yarn syntax. It tries its best but then spits out a comment in the generated yarn block along the lines of "couldn't auto-convert command". I believe it's related to [this issue](https://github.com/facebook/docusaurus/issues/5861) in the docusaurus repo. I marked a TODO comment in the docs tho for us to look into using npm2yarn if/when that is ever resolved

**Review by rwaskiewicz — APPROVED** [review]
LGTM! 🚢

File: versioned_docs/version-v3.0/framework-integration/react.md
Comment (alicewriteswrongs) [inline versioned_docs/version-v3.0/framework-integration/react.md:509]:
```suggestion
If you want your custom elements to work on older browsers, you should call the `applyPolyfills()` function before `defineCustomCustomElements()`. `applyPolyfills()` returns a `Promise`, so you could call it like this:
```

File: versioned_docs/version-v3.0/framework-integration/react.md
Comment (alicewriteswrongs) [inline versioned_docs/version-v3.0/framework-integration/react.md:501]:
```suggestion
In addition, all the Web Components will be automatically defined as the generated component modules are bootstrapped.
```

File: versioned_docs/version-v3.0/framework-integration/react.md
Comment (alicewriteswrongs) [inline versioned_docs/version-v3.0/framework-integration/react.md:418]: I think this should be 'React Router', React doesn't have a single routing solution (or a 'blessed' one):

```suggestion
Ionic Framework can integrate better with React Router.
```

File: docs/framework-integration/react.md
Comment (alicewriteswrongs) [inline docs/framework-integration/react.md:280]: maybe should be

```suggestion
:::tip
```

File: docs/framework-integration/react.md
Comment (alicewriteswrongs) [inline docs/framework-integration/react.md:246]:
```suggestion
add the following:
```

**Review by alicewriteswrongs — CHANGES_REQUESTED** [review]
looks good to me overall, just noticed a few quick text things

**Review by alicewriteswrongs — APPROVED** [review]
lgtm!

**Pull request merged**
